src/core/meshcop/dataset.cpp

```diff
@@ -263,7 +263,7 @@ otError Dataset::Set(const otOperationalDataset &aDataset)
         tlv.Init();
         tlv.SetSeconds(aDataset.mActiveTimestamp);
         tlv.SetTicks(0);
-        Set(tlv);
+        IgnoreError(error = Set(tlv));
```

Comment:
Seems logic changed here. If so, `IgnoreError` is still necessary here?
```suggestion
        error = Set(tlv);
```